html/user/openssl_sign_action.php

```diff
@@ -0,0 +1,148 @@
+<?php
+// This file is part of BOINC.
+// http://boinc.berkeley.edu
+// Copyright (C) 2008 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <http://www.gnu.org/licenses/>.
+
+require_once("../inc/boinc_db.inc");
+require_once("../inc/user.inc");
+require_once("../inc/util.inc");
+//require_once('../inc/recaptchalib.php');
+
+// Needs to be defined, unknown use.
+check_get_args(array("tnow", "ttok"));
+
+// Check the user is online
+$user = get_logged_in_user();
+check_tokens($user->authenticator);
+
+// Optional: Recaptcha support
+//if ($recaptcha_private_key) {
+//    if (!boinc_recaptcha_isValidated($recaptcha_private_key)) {
+//        error_page(
+//            tra("Your reCAPTCHA response was not correct. Please try again.")
+//        );
+//    }
+//}
+
+// Input is passed in from the openssl_sign_form
+// sanitize_tags removes html elements from the user input
+// $user_data = htmlspecialchars((post_str("user_data", true)),ENT_XML1);
```

Comment:
Please remove this as it implies that this is a possibility for projects to use. But it would screw the verification process later when the third party site expects tags in the message. Although if you allow tags in the `user_data` and `msg` fields you would need to encapsulate them with a `CDATA` element in the XML output.

To sumarize: either escape html tags in `user_data` (substitute `>` with `&gt;`) or use something like this: `'<![CDATA['.$message_data.']]>'`

Reply:
I implemented the following which converts special characters to html equivelant:

```$user_data = htmlentities(post_str("user_data", true), ENT_QUOTES, "UTF-8");```

Is that better?